modules/ROOT/pages/notifications/all-notifications.adoc

```diff
@@ -937,6 +938,240 @@ m|WARNING
+====
+
+.Revoking a privilege from a role that does not have that privilege
+====
+Command::
++
+[source, cypher]
+----
+REVOKE WRITE ON GRAPH * FROM editor
+----
+
+Full title of the returned code::
+`REVOKE DENY WRITE ON GRAPH * FROM editor` has no effect.
+
+Description of the returned code::
+The role does not have the privilege. See Status Codes documentation for more information.
+
+Suggestions for improvement::
+Make sure this was the intended privilege and role.
+====
+
+[#_neo_clientnotification_security_impossiblerevokecommand]
+=== Neo.ClientNotification.Security.ImpossibleRevokeCommand
+
+.Notification details
+[cols="<1s,<4"]
+|===
+|Code
+m|Neo.ClientNotification.Security.ImpossibleRevokeCommand
+|Title
+a|`<command>` has no effect.
+|Severity
+m|WARNING
+|Category
+m|SECURITY
+|===
+
+The `<command>` in the title is the part of the command which had no effect.
+This could be either the full command given by the user or a subset of the given command.
+
+.Revoking a non-existing role from a user
+====
+Command::
++
+[source, cypher]
+----
+REVOKE ROLE Foo, reader FROM neo4j
+----
+
+Full title of the returned code::
+`REVOKE ROLE Foo FROM neo4j` has no effect.
+
+Description of the returned code::
+Role does not exist. Make sure nothing is misspelled.
+This notification will become an error in a future major version.
+See Status Codes documentation for more information.
+
+Suggestions for improvement::
+Make sure nothing is misspelled and that this was the intended role and user.
+====
+
+.Revoking a role from a non-existing user
+====
+Command::
++
+[source, cypher]
+----
+REVOKE ROLE reader FROM Foo
+----
+
+Full title of the returned code::
+`REVOKE ROLE reader FROM Foo` has no effect.
+
+Description of the returned code::
+User does not exist. Make sure nothing is misspelled.
+This notification will become an error in a future major version.
+See Status Codes documentation for more information.
+
+Suggestions for improvement::
+Make sure nothing is misspelled and that this was the intended role and user.
+====
+
+.Revoking a privilege from a non-existing role
+====
+Command::
++
+[source, cypher]
+----
+REVOKE GRANT WRITE ON GRAPH * FROM Foo
+----
+
+Full title of the returned code::
+`REVOKE GRANT WRITE ON GRAPH * FROM Foo` has no effect.
+
+Description of the returned code::
+Role does not exist. Make sure nothing is misspelled.
+This notification will become an error in a future major version.
+See Status Codes documentation for more information.
+
+Suggestions for improvement::
+Make sure nothing is misspelled and that this was the intended role.
+====
+
+.Revoking a privilege on a non-existing graph/database from a role
+====
+Command::
++
+[source, cypher]
+----
+REVOKE GRANT WRITE ON GRAPH Foo FROM editor
+----
+
+Full title of the returned code::
+`REVOKE GRANT WRITE ON GRAPH Foo FROM editor` has no effect.
+
+Description of the returned code::
+Database 'Foo' does not exist. Make sure nothing is misspelled.
```

Comment:
The message we currently have for the error on grant/deny have ' and not ` so I'd keep those here
```suggestion
Graph 'Foo' does not exist. Make sure nothing is misspelled.
```

Reply:
I've updated the names to:
- Existing users: `john`, `jane`
- Missing user: `alice`
- Existing roles: `reader`, `admin`, `PUBLIC`
- Missing roles: `manager`
- Existing database/graph: `neo4j`
- Missing database/graph: `neo3j`

I kept graph and database name the same because [that is how we refer to graphs](https://neo4j.com/docs/operations-manual/current/manage-databases/introduction/#manage-databases-concepts):
> Many administrative commands refer to a specific graph by using the database name.